### DISCUSSION TITLE
Installing latest FluidSynth on Raspberry Pi using apt

### DISCUSSION THREAD

**Discussion by albedozero** [FluidSynth/fluidsynth#743] [Q&A] [answered]
I'm mostly interested in an answer to this for educational purposes. I use FluidSynth on a Debian laptop and on a Raspberry Pi, and to get the latest version I just [build it](https://github.com/albedozero/fluidpatcher/wiki/Programs#compile-fluidynth), which works very smoothly - no complaints at all there. However, I notice that there is a .deb file for the latest version (2.1.6) in the [Raspbian repository](http://raspbian.raspberrypi.org/raspbian/pool/main/f/fluidsynth/) - why does `apt-get` only install 1.1.11? Is it something I would need to change in `/etc/apt/sources.list`? I tried downloading the .debs and installing them directly as shown below, but it seems maybe they have dependencies that aren't in the repos yet - but then how am I able to build it from source? I've tried educating myself, but I'm failing - maybe big brains here can help me :)

```
pi@raspberrypi:~ $ sudo apt install ./fluidsynth_2.1.6-1_armhf.deb
Reading package lists... Done
Building dependency tree
Reading state information... Done
Note, selecting 'fluidsynth' instead of './fluidsynth_2.1.6-1_armhf.deb'
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 fluidsynth : Depends: libfluidsynth2 (>= 2.1.0) but it is not installable
              Depends: libsdl2-2.0-0 (>= 2.0.12+dfsg1) but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
pi@raspberrypi:~ $ sudo apt install ./libfluidsynth2_2.1.6-1_armhf.deb
Reading package lists... Done
Building dependency tree
Reading state information... Done
Note, selecting 'libfluidsynth2' instead of './libfluidsynth2_2.1.6-1_armhf.deb'
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 libfluidsynth2 : Depends: libc6 (>= 2.29) but 2.28-10+rpi1 is to be installed
                  Depends: libinstpatch-1.0-2 (>= 1.1.2) but it is not installable
                  Depends: libreadline8 (>= 6.0) but it is not installable
                  Depends: libsdl2-2.0-0 (>= 2.0.12+dfsg1) but it is not going to be installed
                  Depends: timgm6mb-soundfont (>= 1.3-3~) but it is not going to be installed or
                           sf3-soundfont-gm but it is not installable
E: Unable to correct problems, you have held broken packages.
```

**Comment by derselbst** [comment, marked as answer]
> why does apt-get only install 1.1.11?

It looks like you're running the stable distro, hence only the stable repository is considered. According to the [packaging status](https://github.com/FluidSynth/fluidsynth/wiki/Download), 2.1.6 and all of its "missing" dependencies are part of the testing repository, so you need to figure out how to convince apt-get to install it from the testing repo.
